### ISSUE TITLE
[DOMLauncher] add_static_property confuses sphinx (Trac #1740)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1732]
The pybindings of `icecube.DOMLauncher` use `.add_static_property()` for three properties: `I3DOM.merge_pulses`, `I3DOM.use_drooped_template`, and `I3DOM.use_tabulation`. Unfortunately, this confuses sphinx documentation because`merge_pulses`is in `I3DOM`'s attribute list but `I3DOM.merge_pulses.__doc__` is unavailable. 

The following error is spit out by sphinx:

```text
attribute u'I3DOM.merge_pulses' from module u'icecube.DOMLauncher'; the following exception was raised:
Traceback (most recent call last):
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 392, in import_object
    obj = self.get_attr(obj, part)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 288, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py", line 115, in safe_getattr
    raise AttributeError(name)
AttributeError: merge_pulses
/build/buildslave/kipple_cvmfs/docs/build/sphinx_src/source/python/icecube.DOMLauncher.rst:4: WARNING: autodoc: failed to import attribute u'I3DOM.use_drooped_template' from module u'icecube.DOMLauncher'; the following exception was raised:
Traceback (most recent call last):
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 392, in import_object
    obj = self.get_attr(obj, part)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 288, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py", line 115, in safe_getattr
    raise AttributeError(name)
AttributeError: use_drooped_template
/build/buildslave/kipple_cvmfs/docs/build/sphinx_src/source/python/icecube.DOMLauncher.rst:4: WARNING: autodoc: failed to import attribute u'I3DOM.use_tabulation' from module u'icecube.DOMLauncher'; the following exception was raised:
Traceback (most recent call last):
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 392, in import_object
    obj = self.get_attr(obj, part)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 288, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py", line 115, in safe_getattr
    raise AttributeError(name)
AttributeError: use_tabulation
```

Migrated from https://code.icecube.wisc.edu/ticket/1740
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:38", 
    "description": "The pybindings of `icecube.DOMLauncher` use `.add_static_property()` for three properties: `I3DOM.merge_pulses`, `I3DOM.use_drooped_template`, and `I3DOM.use_tabulation`. Unfortunately, this confuses sphinx documentation because`merge_pulses`is in `I3DOM`'s attribute list but `I3DOM.merge_pulses.__doc__` is unavailable. \n\nThe following error is spit out by sphinx:\n\n{{{\nattribute u'I3DOM.merge_pulses' from module u'icecube.DOMLauncher'; the following exception was raised:\nTraceback (most recent call last):\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 392, in import_object\n    obj = self.get_attr(obj, part)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 288, in get_attr\n    return safe_getattr(obj, name, *defargs)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py\", line 115, in safe_getattr\n    raise AttributeError(name)\nAttributeError: merge_pulses\n/build/buildslave/kipple_cvmfs/docs/build/sphinx_src/source/python/icecube.DOMLauncher.rst:4: WARNING: autodoc: failed to import attribute u'I3DOM.use_drooped_template' from module u'icecube.DOMLauncher'; the following exception was raised:\nTraceback (most recent call last):\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 392, in import_object\n    obj = self.get_attr(obj, part)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 288, in get_attr\n    return safe_getattr(obj, name, *defargs)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py\", line 115, in safe_getattr\n    raise AttributeError(name)\nAttributeError: use_drooped_template\n/build/buildslave/kipple_cvmfs/docs/build/sphinx_src/source/python/icecube.DOMLauncher.rst:4: WARNING: autodoc: failed to import attribute u'I3DOM.use_tabulation' from module u'icecube.DOMLauncher'; the following exception was raised:\nTraceback (most recent call last):\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 392, in import_object\n    obj = self.get_attr(obj, part)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/ext/autodoc.py\", line 288, in get_attr\n    return safe_getattr(obj, name, *defargs)\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v1/RHEL_6_x86_64/lib/python2.7/site-packages/sphinx/util/inspect.py\", line 115, in safe_getattr\n    raise AttributeError(name)\nAttributeError: use_tabulation\n}}}", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067158057333", 
    "component": "combo simulation", 
    "summary": "[DOMLauncher] add_static_property confuses sphinx", 
    "priority": "normal", 
    "keywords": "documentation", 
    "time": "2016-06-10T09:57:06", 
    "milestone": "", 
    "owner": "sflis", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160610T14:46:53: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "sflis"

**Comment by icecube-trac** [comment]
Trac update at 20160615T18:39:24: olivas changed component from "cmake" to "simulation"

**Comment by icecube-trac** [comment]
Trac update at 20160814T18:02:05: olivas changed summary from "[pybindings] add_static_property confuses sphinx" to "[DOMLauncher] add_static_property confuses sphinx"

**Comment by icecube-trac** [comment]
Trac update at 20161116T10:48:35: 
* kjmeagher commented:

> In changeset 151537:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="151537"
> I can't figure out why they pybindings for these were using `add_static_property` in the first place, they do not bind to static methods, fixes #1732
> ```


* kjmeagher changed resolution from "" to "fixed"
* kjmeagher changed status from "assigned" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:38: 
* olivas commented:

> Milestone August 2016 deleted


* olivas changed milestone from "August 2016" to ""
